diesel_codegen/src/schema_inference/mod.rs

```diff
@@ -124,60 +118,143 @@ fn next_str_lit<T: Iterator<Item=P<ast::Expr>>>(
     }
 }
 
-fn get_table_data(conn: &PgConnection, table_name: &str) -> QueryResult<Vec<PgAttr>> {
-    use self::data_structures::pg_attribute::dsl::*;
-    use self::data_structures::pg_type::dsl::{pg_type, typname};
-    let t_oid = try!(table_oid(conn, table_name));
 
-    pg_attribute.inner_join(pg_type)
-        .select((attname, typname, attnotnull))
-        .filter(attrelid.eq(t_oid))
-        .filter(attnum.gt(0).and(attisdropped.ne(true)))
-        .order(attnum)
-        .load(conn)
+fn column_def_tokens(cx: &mut ExtCtxt, attr: &ColumnInformation, conn: &InferConnection)
+    -> Vec<ast::TokenTree>
+{
+    let column_name = str_to_ident(&attr.column_name);
+    let tpe = determine_column_type(cx, attr, conn);
+    quote_tokens!(cx, $column_name -> $tpe,)
 }
 
-fn table_oid(conn: &PgConnection, table_name: &str) -> QueryResult<u32> {
-    use self::data_structures::pg_class::dsl::*;
-    pg_class.select(oid).filter(relname.eq(table_name)).first(conn)
+
+// FIXME: Remove the duplicates of this function once expression level attributes
+// are stable (I believe this is in 1.7)
+#[cfg(all(feature = "sqlite", not(feature = "postgres")))]
+fn establish_connection(
+    cx: &mut ExtCtxt,
+    sp: Span,
+    database_url: &str,
+) -> Result<InferConnection, Box<MacResult>>{
```

Comment:
Please ensure that your code matches the style guide (spaces before all your braces)